apollo-router/src/spec/schema.rs

```diff
@@ -116,8 +116,21 @@ impl Schema {
                 if url.is_empty() {
                     return Err(SchemaError::MissingSubgraphUrl(name.to_string()));
                 }
-                let url = Uri::from_str(url)
-                    .map_err(|err| SchemaError::UrlParse(name.to_string(), err))?;
+                // there is noi standard for unix socket URLs apparently
+                let url = if url.starts_with("unix://") {
+                    println!("authority: {}", &url[7..]);
+                    // there is no specified format for unix socket URLs (cf https://github.com/whatwg/url/issues/577)
+                    // so a unix:// URL will not be parsed by http::Uri
+                    // To fix that, hyperlocal came up with its own Uri type that can be converted to http::Uri.
+                    // It hides the socket path in a hex encoded authority that the unix socket connector will
+                    // know how to decode
+                    let u = hyperlocal::Uri::new(&url[7..], "/").into();
+                    println!("generated Uri = {u:?} from '{url}'");
```

Comment:
```suggestion
```

Reply:
fixed in [512b54b](https://github.com/apollographql/router/pull/4757/commits/512b54b3a36c6f13168081965eb6a33052b0d482)